### DISCUSSION TITLE
Fine-tuning of multiple speakers

### DISCUSSION THREAD

**Discussion by todochenxi** [yl4579/StyleTTS2#157] [Q&A]
你好，我想咨询一下使用多个speaker的数据去微调模型的流程和结果，我疑惑的点如下：
1. 数据集的trin_list.txt 的格式是什么样的，
2. config_ft.yml 中root_path要如何配置呢
3. 如果我是用三个speaker的数据去微调，得到的模型可以指定speaker去推理嘛
4. 假设可以的话，那是否可以在单个speaker的基础上去扩展，添加另外的声音呢。。。

Hello, I would like to ask about the process and results of using data from multiple speakers to fine-tune the model. My doubts are as follows:
1. What is the format of trin_list.txt of the data set?
2. How to configure root_path in config_ft.yml
3. If I use the data of three speakers for fine-tuning, can the resulting model specify the speaker for inference?
4. If possible, can it be expanded based on a single speaker to add additional sounds. . .

**Comment by todochenxi** [comment]
Oh, I found OOD_texts.txt, I think 1 and 2 have been solved :)

**Comment by todochenxi** [comment]
Oh, for 3, I suddenly thought that it can be achieved by just changing the reference audio. . .

**Comment by todochenxi** [comment]
For 4, can we continue to fine-tune based on the previously fine-tuned model? Or do you have to retrain every time a new character is added? . .

**Reply by yl4579** [reply]
Yes, you can finetune previous models if you want to add more voices, if you still have the data from previous speakers. If you don't, the model will forget previous speakers depending on how long you train it. It is called catastrophic forgetting and is an active research area called "continual learning". It can be used by generating speech of that speaker if you don't have access to the dataset anymore.

**Reply by todochenxi** [reply]
Thank you for your reply. Last Friday I did an experiment to fine-tune the data sets of two speakers. By switching reference audio, I achieved the purpose of switching speakers. The results are stunning and the work has so much potential. Thank you for your work, salute ~ :)

**Reply by junylee11** [reply]
How long was the data used for fine tuning?

**Reply by todochenxi** [reply]
30m

**Reply by todochenxi** [reply]
I just tried it briefly, you can customize your data

**Reply by junylee11** [reply]
Thank you!
